### ISSUE TITLE
Domino Java Application Template

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Moonshine-IDE#903]
I worked with @JustinProminic to setup a Domino API example starting with the Java Gradle template.  This took longer than expected, since it required additional configuration for the external library and classpath.  This is an important use case for the Domino support we would like to add to Moonshine, so we want to add a new template for this.

I spent some more time making a clean example project, and I put it on GitHub here.  I'll make this public once I have a chance to review it with Justin:
https://github.com/Moonshine-IDE/DominoAPILocalExample

Some Notes:
- The project name will need to be replaced in multiple locations in build.gradle and RunJAR.sh 
- I'd also like to populate the configured Notes path (and JDK 8 path?) in build.gradle and RunJAR.sh
- I still need to review this on Windows and create a build script.

This is not immediately urgent, so I have assigned this to a future milestone.

**Comment by JoelProminic** [comment]
Note that I currently define the `DYLD_LIBRARY_PATH` environment variable in build.gradle.   Once we add this environment variable in the environment script for #930, I should remove/disable this workaround in the template.  I may keep the `runApp` task with a commented-out example of setting the environment in case it is needed.

**Comment by JoelProminic** [comment]
I made the  DominoAPILocalExample project public.  I don't see anything here that could cause a problem.  I should set a license, though.

**Comment by rat-moonshine** [comment]
![image](https://user-images.githubusercontent.com/24370870/157207016-71caa0c7-8703-4305-ae8c-ae87d5a18e55.png)
![image](https://user-images.githubusercontent.com/24370870/157207063-2f9b6a7a-ba6e-4273-8621-ddd810c6b1d5.png)

This can be tested with macOS Bamboo branch-build plan.

**Comment by JoelProminic** [comment]
This is looking pretty good so far, but there are a couple things I meant to clarify.
- README.md should probably be excluded - this is written for the Git project, so some of this information won't make sense.   There are some parts that may be useful for reference, though - I'll think about whether I want to modify the README.md instead
- We need to replace the value for `notesInstallation` in build.gradle with the "HCL Notes Installation" setting
- Similarly, Moonshine needs to update RunJar.sh:
    - `DYLD_LIBRARY_PATH`:  HCL Notes Installation
    - `JAVA_HOME`:  needs to be updated to the project JDK (which will always be JDK 8

**Comment by JoelProminic** [comment]
The paths should be correct for Windows as well, but I need to update `PATH` as well.   I'll let you know when I have the script updated.

Also, I see the path has changed for HCL Notes 12.0.1:  `/Applications/HCL Notes.app/Contents/Resources/jvm/lib/ext/Notes.jar` (`Contents/Resources` instead of `Contents/MacOS`).  I can handle this with some Groovy code.

**Comment by rat-moonshine** [comment]
- README.md now removed from template files
- `notesInstallation` updating with HCL Notes path in `build.gradle`
- `DYLD_LIBRARY_PATH` updating with HCL Notes path in `RunJar.sh`
- `JAVA_HOME` updating with JDK path in `RunJar.sh`

A few things to note, however:

1. What value the placeholder should hold in case HCL Notes path, or JDK 8 path is not setup in Moonshine?
2. I was not clear with Windows' reference [above](https://github.com/Moonshine-IDE/Moonshine-IDE/issues/903#issuecomment-1062403435). Moonshine stores Notes executable path, following usage from `build.gradle`could be problematic in that case:
```
String notesInstallation = 'C:\Program Files\...\Notes.exe'
String notesJarPath = "$notesInstallation/jvm/lib/ext/Notes.jar"
```
(My Windows system is broken, nor I had Notes in my Windows PC)

**Comment by JoelProminic** [comment]
The HCL Notes Installation path looks like this on Windows, with a fresh Notes 12.0.1 install:  `C:\Program Files (x86)\HCL Notes`
![image](https://user-images.githubusercontent.com/24532731/157483313-45042648-c813-4c94-9b92-6a623659f1e9.png)

So this should work fine with the current logic.   However, I forgot to mention that the backslash path separators should be replaced with forward slashes in `build.gradle`.  Doubling the backslashes will also work, but that is not consistent with the `notesJarPath` line.

Thinking about this more, updating the paths in the files makes the projects less portable.  I'm looking into reading these settings from environment variables. but I remember this was a problem for macOS and Gradle...

**Comment by rat-moonshine** [comment]
I have added some handling to determine between Gradle, Maven and Domino project types.

I also updated templating generation based on [above](https://github.com/Moonshine-IDE/Moonshine-IDE/issues/903#issuecomment-1063106572) observation.

**Comment by JoelProminic** [comment]
The template looks good now, and it runs properly on macOS Catalina with Notes 11.   We should do some tests with Windows and macOS Mojave as well, but I think that can wait until after a merge.

I'm planning to make some updates to the template, but these shouldn't require any update to the logic.   I think it should be fine to  merge this now.

**Comment by rat-moonshine** [comment]
This issue is now merged into `master`.

**Comment by JoelProminic** [comment]
This is currently waiting on a couple tweaks to the build.gradle:
- Windows:  need to update PATH instead of setting DYLD_LIBRARY_PATH.  This wasn't working in my last tests in [this issue](https://github.com/Moonshine-IDE/DominoAPILocalExample/issues/1), though.
- macOS Monterey support.  See [this comment](https://github.com/Moonshine-IDE/Moonshine-IDE/issues/903#issuecomment-1062403435)

**Comment by JoelProminic** [comment]
I made some updates for build.gradle [here](https://github.com/Moonshine-IDE/DominoAPILocalExample/blob/9cc7f231310e228591fd152eeb7c9640d67b9c81/build.gradle).  This will need to be updated in the template.

**Comment by JoelProminic** [comment]
I made a quick update for this myself.

I confirmed this is working now on macOS Catalina.  @Aszusz or @JesseProminic, can you test Java Domino projects on your macOS workstations to see if it works without modification?

For Windows, this will fail with a 32-bit error.  I don't have an immediate solution for this, so we may revisit it in a later release.
